platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1069,12 +1069,13 @@ - (id)mgl_jsonExpressionObject {
         }
             
         case NSKeyPathExpressionType: {
-            NSArray *expressionObject;
-            for (NSString *pathComponent in self.keyPath.pathComponents.reverseObjectEnumerator) {
+            NSMutableArray *expressionObject;
+            NSArray *keyPath = [self.keyPath componentsSeparatedByString:@"."];
+            for (NSString *pathComponent in keyPath.reverseObjectEnumerator) {
                 if (expressionObject) {
-                    expressionObject = @[@"get", pathComponent, expressionObject];
+                    [expressionObject addObject:@[@"get", pathComponent]];
```

Comment:
`map.box.gl` should become `["get", "gl", ["get", "box", ["get", "map"]]]`, not `["get", "gl", ["get", "box"], ["get", "map"]]`.

Reply:
That code already handles that case and returns `["get", "gl", ["get", "box", ["get", "map"]]]` I added a test case for that:
https://github.com/mapbox/mapbox-gl-native/pull/11959/files#diff-a4216b00e3cfddef98fc22cbc71f3116R316